### PR TITLE
Updated tests for FilterFunction

### PR DESCRIPTION
**Pull request by Aakanksha0311**
[Insert JIRA Issue number and update link](https://issues.redhat.com/browse/MGDOBR-)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.

</details>

### CODE REVIEW

**Comment by sutaakar** [comment]
`/rebase`

File: integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/FilterSteps.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    public FilterSteps(TestContext context) {
+        this.context = context;
+    }
+
+
+    @When("^send a cloud event to the Ingress of the Bridge \"([^\"]*)\":$")
+    public void sendCloudEventToIngressOfBridgeWithPath(String testBridgeName, String cloudEvent) {
+        IngressSteps ingressSteps=new IngressSteps(context);
+        ingressSteps.sendAndCheckCloudEvent(testBridgeName, cloudEvent, "", 200);
+    }
```
Comment (sutaakar) [inline integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/FilterSteps.java:33]: This is a duplication of https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/IngressSteps.java#L63-L66
Can be removed.

File: integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/IngressSteps.java [hunk 2]
```diff
@@ -70,7 +70,7 @@ public void sendCloudEventToIngressOfBridgeWithPathAndDefaultHeaders(String test
         sendAndCheckCloudEventWithHeaders(testBridgeName, cloudEvent, path, parseHeaders(headers), 200);
     }
 
-    private void sendAndCheckCloudEvent(String testBridgeName, String cloudEvent, String path, int responseCode) {
+    void sendAndCheckCloudEvent(String testBridgeName, String cloudEvent, String path, int responseCode) {
```
Comment (sutaakar) [inline integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/IngressSteps.java:73]: This change is not needed then

File: integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/FilterSteps.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+
+
+
+    @Then("^Webhook site contains request with text \"([^\"]*)\" within (\\d+) (?:minute|minutes)$")
+    public void webhookSiteContainsRequest(String requestText, int timeoutMinutes) {
```
Comment (sutaakar) [inline integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/FilterSteps.java:39]: This is a duplication of https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/WebhookSiteSteps.java#L24
Can be removed

File: integration-tests/src/test/resources/features/filter.feature [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  },
+  "filters": [
+    {
+      "key": "source",
+      "type": "StringEquals",
+      "value": "StorageService"
+    }
+  ]
+}
+    """
```
Comment (sutaakar) [inline integration-tests/src/test/resources/features/filter.feature:24]: Please adjust alignment, see for example https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/integration-tests/src/test/resources/features/action-webhook.feature#L11-L22

File: integration-tests/src/test/resources/features/filter.feature [hunk 5]
```diff
@@ -0,0 +1,42 @@
+Feature: Filter tests
+
+  Background:
+    Given authenticate against Manager
+    And create a new Bridge "mybridge"
+    And the Bridge "mybridge" is existing with status "ready" within 4 minutes
+    And the Ingress of Bridge "mybridge" is available within 2 minutes
+    And add a Processor to the Bridge "mybridge" with body:
```
Comment (sutaakar) [inline integration-tests/src/test/resources/features/filter.feature:8]: I would put this step as part of scenario as it contains the specific scenario logic.

Comment (Aakanksha0311) [reply integration-tests/src/test/resources/features/filter.feature:8]: Okay! Will do the necessary changes! Thanks

**Comment by Aakanksha0311** [comment]
Unable to push to this branch due to conflicts....created a new PR
[https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/702]

**Pull request closed without merging**